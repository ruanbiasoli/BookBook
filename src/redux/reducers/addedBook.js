import { POST_BOOK } from "../actions";

const defaultState = {
  book: [],
};
const userBookChoice = (state = defaultState, action) => {
  switch (action.type) {
    case POST_BOOK:
      return {
        ...state,
        book: {
          data: action.data,
        },
      };
    default:
      return state;
  }
};

export default userBookChoice;
