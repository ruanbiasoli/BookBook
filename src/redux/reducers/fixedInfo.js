import { FIXED_BOOK } from "../actions";

const defaultState = {
  fixedInfo: [],
  fixedCategory: [],
};
const fixedInfo = (state = defaultState, action) => {
  switch (action.type) {
    case FIXED_BOOK:
      return {
        ...state,
        fixedCategory: action.imageCategory,
        fixedInfo: action.data.items.map(({ volumeInfo, id }) => ({
          title: volumeInfo.title !== undefined ? volumeInfo.title : "",
          authors:
            volumeInfo.authors !== undefined
              ? volumeInfo.authors.join(",")
              : "",
          categories:
            volumeInfo.categories !== undefined
              ? volumeInfo.categories.join(",")
              : "",
          description:
            volumeInfo.description !== undefined
              ? volumeInfo.description.split(".")[0]
              : "",
          image:
            volumeInfo.imageLinks !== undefined
              ? volumeInfo.imageLinks.thumbnail.replace("http", "https")
              : "https://books.google.com/books/content?id=AAWDRDwd3p0C&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
          googleID: id !== undefined ? id : "",
        })),
      };
    default:
      return state;
  }
};

export default fixedInfo;
