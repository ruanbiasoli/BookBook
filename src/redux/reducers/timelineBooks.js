import { TIMELINE } from "../actions";

const defaultState = {
  books: [],
};
const timelineBooks = (state = defaultState, action) => {
  switch (action.type) {
    case TIMELINE:
      return {
        ...state,
        books: action.data.map(
          ({ title, author, categories, image_url, creator }) => ({
            title: title,
            author: author,
            categories: categories,
            image:
              image_url !== null
                ? image_url.replace("http", "https").replace("httpss", "https")
                : "https://books.google.com/books/content?id=AAWDRDwd3p0C&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
            username: creator.user,
            email: creator.email,
          })
        ),
      };
    default:
      return state;
  }
};

export default timelineBooks;
