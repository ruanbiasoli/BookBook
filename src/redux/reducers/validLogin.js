import { LOGIN, LOGOUT } from "../actions";

const defaultState = {
  user: JSON.parse(localStorage.getItem("user")),
  token: null,
};
const validLogin = (state = defaultState, action) => {
  switch (action.type) {
    case LOGIN:
      return {
        ...state,
        user: action.data,
        token: action.token,
      };
    case LOGOUT:
      return {
        ...state,
        user: null,
        token: null,
      };
    default:
      return state;
  }
};

export default validLogin;
