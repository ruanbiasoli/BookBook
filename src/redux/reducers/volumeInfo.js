import { ACTION_BOOK } from "../actions";

const defaultState = {
  volumeInfo: [],
  data: [],
};
const volumeInfo = (state = defaultState, action) => {
  switch (action.type) {
    case ACTION_BOOK:
      return {
        ...state,
        data: action.data.items,
        volumeInfo: action.data.items.map(({ volumeInfo, id }) => ({
          title: volumeInfo.title !== undefined ? volumeInfo.title : "",
          authors:
            volumeInfo.authors !== undefined
              ? volumeInfo.authors.join(",")
              : "",
          categories:
            volumeInfo.categories !== undefined
              ? volumeInfo.categories.join(",")
              : "",
          description:
            volumeInfo.description !== undefined
              ? volumeInfo.description.split(".")[0]
              : "",
          image:
            volumeInfo.imageLinks !== undefined
              ? volumeInfo.imageLinks.thumbnail.replace("http", "https")
              : "https://books.google.com/books/content?id=AAWDRDwd3p0C&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
          googleID: id !== undefined ? id : "",
        })),
      };
    default:
      return state;
  }
};

export default volumeInfo;
