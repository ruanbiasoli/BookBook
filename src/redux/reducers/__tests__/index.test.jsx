import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
// import { useDispatch } from 'react-redux';
import validLogin from '../validLogin'
import addedBook from '../addedBook';
import fixedInfo from '../fixedInfo';
import timelineBooks from '../timelineBooks';
import volumeInfo from '../volumeInfo';

describe('snapshot', () => {
    it('valid login test - must be a function', () => {
        const testAction = renderer
        .create(<button onClick={validLogin}>test</button>)

          expect(testAction).toMatchSnapshot();
    })

    it('addedBook test - must be a function', () => {
        const addedBookTest = renderer
        .create(<button onClick={addedBook}>test</button>)

          expect(addedBookTest).toMatchSnapshot();
    })

    it('fixedInfo test - must be a function', () => {
        const testAction = renderer
        .create(<button onClick={fixedInfo}>test</button>)

          expect(testAction).toMatchSnapshot();
    })

    it('timelineBooks test - must be a function', () => {
        const testAction = renderer
        .create(<button onClick={timelineBooks}>test</button>)

          expect(testAction).toMatchSnapshot();
    })

    it('volumeInfo test - must be a function', () => {
        const testAction = renderer
        .create(<button onClick={volumeInfo}>test</button>)

          expect(testAction).toMatchSnapshot();
    })
})

