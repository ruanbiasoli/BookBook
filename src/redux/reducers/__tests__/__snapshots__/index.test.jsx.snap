// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot addedBook test - must be a function 1`] = `
<button
  onClick={[Function]}
>
  test
</button>
`;

exports[`snapshot fixedInfo test - must be a function 1`] = `
<button
  onClick={[Function]}
>
  test
</button>
`;

exports[`snapshot timelineBooks test - must be a function 1`] = `
<button
  onClick={[Function]}
>
  test
</button>
`;

exports[`snapshot valid login test - must be a function 1`] = `
<button
  onClick={[Function]}
>
  test
</button>
`;

exports[`snapshot volumeInfo test - must be a function 1`] = `
<button
  onClick={[Function]}
>
  test
</button>
`;
